### PR TITLE
Add registration and login

### PR DESCRIPTION
**Pull request by bartoszbialecki**

### CODE REVIEW

File: controllers/auth.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+  const isEmailExist = await User.findOne({ email: req.body.email })
+
+  if (isEmailExist) {
+    return res
+      .status(400)
+      .json({ error: 'An account with this email address already exists.' })
+  }
+
+  const user = new User({ ...req.body })
```
Comment (pdybowski) [inline controllers/auth.js]: (req.body) should work

File: controllers/auth.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+  try {
+    const savedUser = await user.save()
```
Comment (pdybowski) [inline controllers/auth.js:43]: we dont need result here

File: controllers/auth.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+  return jwt.sign(payload, secret, options)
+}
+
+const responseWithToken = (res, user) => {
+  const token = createToken(user)
+
+  res.header('X-Auth-Token', token).json({
+    error: null,
+    data: {
```
Comment (pdybowski) [inline controllers/auth.js:27]: can we pass user role?

File: controllers/auth.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+      userEmail: user.email,
+    },
+  })
+}
+
+exports.register = async (req, res) => {
+  const { error } = validateCreateUser(req.body)
+
+  if (error) {
+    return res.status(400).json({ error: error.details[0].message })
```
Comment (pdybowski) [inline controllers/auth.js]: remove json and set send(message) like in others PRs. we need to make it the same in every place to make front work easier

File: routes/auth.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+router.post('/register', async (req, res) => {
+  authController.register(req, res)
+})
+
+router.post('/login', async (req, res) => {
+  authController.login(req, res)
+})
+
+// TODO: example how to use verifyToken, delete later
```
Comment (pdybowski) [inline routes/auth.js]: you can remove it. we should know how to use middleware functions

File: startup/routes.js [hunk 6]
```diff
@@ -16,7 +16,7 @@ module.exports = function (app) {
   app.use(express.json())
 
   app.use('/api', limit)
-  app.use('/api/v1/users', example)
+  app.use('/api/user', authRoute)
```
Comment (pdybowski) [inline startup/routes.js]: user will be used for user routes
change path to /api/auth, and as second argument set just auth not authRoute

File: controllers/auth.js (hunk 2 above)
Comment (bartoszbialecki) [reply controllers/auth.js:43]: why, we need on frontend to have userId to get user info

File: controllers/auth.js (hunk 4 above)
Comment (bartoszbialecki) [reply controllers/auth.js]: every response should be in json format

File: controllers/auth.js (hunk 2 above)
Comment (pdybowski) [reply controllers/auth.js:43]: okay, but where do we use this variable?

File: controllers/auth.js (hunk 4 above)
Comment (pdybowski) [reply controllers/auth.js]: why?

Comment (pdybowski) [reply controllers/auth.js]: we pass only message so on front we will just receive message - this is what we want

Comment (bartoszbialecki) [reply controllers/auth.js]: why? This is not standard, we should work with json and on frontend parse json

File: controllers/auth.js (hunk 3 above)
Comment (bartoszbialecki) [reply controllers/auth.js:27]: but what for? On the fronted we should, I think, read all data we need for user with another api call or not? Or should I return all user data here?

File: controllers/auth.js (hunk 4 above)
Comment (pdybowski) [reply controllers/auth.js]: this is additional work to be done. I checked blogs and some devs send(error), others send(status, {message: error}) and others used json. So this is just the convention and we use send(error) in other places

Comment (bartoszbialecki) [reply controllers/auth.js]: This is no work and what if you want return more data, than simple message?

Comment (pdybowski) [reply controllers/auth.js]: okay leave it with json, probably we will send more properties there

File: controllers/auth.js (hunk 3 above)
Comment (pdybowski) [reply controllers/auth.js:27]: we have some kind of navigation. For hotel owner there will be additional button 'settings' etc...
For user there will be no such a thing. 
For admin there will be 'admin panel' button.

We want to allow hotel owner make reservations

File: controllers/auth.js (hunk 4 above)
Comment (pdybowski) [reply controllers/auth.js]: but change error object name to message

File: controllers/auth.js (hunk 3 above)
Comment (bartoszbialecki) [reply controllers/auth.js:27]: yes, I know, I mean that when I logged in I can call another request, e.g. /api/user/1 and get all information I need or if you want I can all data return from login or register endpoint but then I should return all data we could need in the future

File: controllers/auth.js (hunk 4 above)
Comment (pdybowski) [reply controllers/auth.js]: on front we will catch(error) and error.message looks better then error.error

File: controllers/auth.js (hunk 3 above)
Comment (pdybowski) [reply controllers/auth.js:27]: acctually when login or register we can store only id, email and role. id for other requests, role for nav buttons and email hmmm Im not even sure that we need email there
reservations will be get in other request when we click to the reservation tab.

Comment (bartoszbialecki) [reply controllers/auth.js:27]: in many places you can need those data, e.g. first name and last name by paying, or in settings when user want update his data etc.

Comment (pdybowski) [reply controllers/auth.js:27]: oh okay, youre right, so in response pass everything

Comment (pdybowski) [reply controllers/auth.js:27]: or run other enpoint /userInfo with all user properties

**Pull request merged**